submissions/Oleksii_Drohachov/OOP-exercise/index.js

```diff
@@ -0,0 +1,98 @@
+class Creature {
+  constructor(name, gender, saying, friends) {
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+    this.friends = friends;
+    this.propsArray = [
+      "species",
+      "name",
+      "gender",
+      "saying",
+      "friends",
+      "legs",
+    ];
+  }
+  output = () => {
+    return this.propsArray
+      .map((property) =>
+        Array.isArray(this[property])
+          ? this[property].join(", ")
+          : this[property]
+      )
+      .join("; ");
+  };
+}
+
+class Animal extends Creature {
+  constructor(name, gender, saying, friends) {
+    super(name, gender, saying, friends);
+    this.legs = 4;
+  }
+}
+
+class Human extends Creature {
+  constructor(name, gender, saying, friends) {
+    super(name, gender, saying, friends);
+    this.species = "human";
+    this.legs = 2;
```

Comment:
`legs` looks like a shared property. Should be generalized to a base class just like any other shared property.
Same refers to `species`.

The criteria of generalization is property per se. It can bear different values for different class instances.